## Data Analysis 2
## Knowledge Check 1 

Import pandas and requests library to pull in data from Makeup API.

In [1]:
import pandas as pd
import requests

In [2]:
# get makeup api and print the message
makeup = requests.get('http://makeup-api.herokuapp.com/api/v1/products.json')
if makeup:
    print('Request is successful.')
else:
    print('Request returned an error.')

Request is successful.


In [3]:
# print the data in json format
makeup.json()[0]

{'id': 1048,
 'brand': 'colourpop',
 'name': 'Lippie Pencil',
 'price': '5.0',
 'price_sign': '$',
 'currency': 'CAD',
 'image_link': 'https://cdn.shopify.com/s/files/1/1338/0845/collections/lippie-pencil_grande.jpg?v=1512588769',
 'product_link': 'https://colourpop.com/collections/lippie-pencil',
 'website_link': 'https://colourpop.com',
 'description': 'Lippie Pencil A long-wearing and high-intensity lip pencil that glides on easily and prevents feathering. Many of our Lippie Stix have a coordinating Lippie Pencil designed to compliment it perfectly, but feel free to mix and match!',
 'rating': None,
 'category': 'pencil',
 'product_type': 'lip_liner',
 'tag_list': ['cruelty free', 'Vegan'],
 'created_at': '2018-07-08T23:45:08.056Z',
 'updated_at': '2018-07-09T00:53:23.301Z',
 'product_api_url': 'http://makeup-api.herokuapp.com/api/v1/products/1048.json',
 'api_featured_image': '//s3.amazonaws.com/donovanbailey/products/api_featured_images/000/001/048/original/open-uri20180708-4-13ok

In [4]:
# to see the type of data
print(type(makeup.json()))


<class 'list'>


Using pandas built in function 'json_normalize' to flatten our data.

In [5]:
# normalize and  flatten the json data into a dataframe
makeup_api = pd.json_normalize(makeup.json())
makeup_api.head(4) # print the first 4 rows of the dataframe

id      brand                  name price price_sign currency  \
0  1048  colourpop         Lippie Pencil   5.0          $      CAD   
1  1047  colourpop           Blotted Lip   5.5          $      CAD   
2  1046  colourpop           Lippie Stix   5.5          $      CAD   
3  1045  colourpop  No Filter Foundation  12.0          $      CAD   

                                          image_link  \
0  https://cdn.shopify.com/s/files/1/1338/0845/co...   
1  https://cdn.shopify.com/s/files/1/1338/0845/pr...   
2  https://cdn.shopify.com/s/files/1/1338/0845/co...   
3  https://cdn.shopify.com/s/files/1/1338/0845/pr...   

                                        product_link  \
0    https://colourpop.com/collections/lippie-pencil   
1  https://colourpop.com/collections/lippie-stix?...   
2      https://colourpop.com/collections/lippie-stix   
3  https://colourpop.com/products/no-filter-matte...   

                                        website_link  \
0                              https://colourpop.com   
1                              https://colourpop.com   
2                              https://colourpop.com   
3  https://colourpop.com/products/no-filter-matte...   

                                         description  rating  category  \
0  Lippie Pencil A long-wearing and high-intensit...     NaN    pencil   
1  Blotted Lip Sheer matte lipstick that creates ...     NaN  lipstick   
2  Lippie Stix Formula contains Vitamin E, Mango,...     NaN  lipstick   
3  Developed for the Selfie Age, our buildable fu...     NaN    liquid   

  product_type               tag_list                created_at  \
0    lip_liner  [cruelty free, Vegan]  2018-07-08T23:45:08.056Z   
1     lipstick  [cruelty free, Vegan]  2018-07-08T22:01:20.178Z   
2     lipstick  [cruelty free, Vegan]  2018-07-08T21:47:49.858Z   
3   foundation  [cruelty free, Vegan]  2018-07-08T18:22:25.273Z   

                 updated_at  \
0  2018-07-09T00:53:23.301Z   
1  2018-07-09T00:53:23.287Z   
2  2018-07-09T00:53:23.274Z   
3  2018-07-09T00:53:23.313Z   

                                     product_api_url  \
0  http://makeup-api.herokuapp.com/api/v1/product...   
1  http://makeup-api.herokuapp.com/api/v1/product...   
2  http://makeup-api.herokuapp.com/api/v1/product...   
3  http://makeup-api.herokuapp.com/api/v1/product...   

                                  api_featured_image  \
0  //s3.amazonaws.com/donovanbailey/products/api_...   
1  //s3.amazonaws.com/donovanbailey/products/api_...   
2  //s3.amazonaws.com/donovanbailey/products/api_...   
3  //s3.amazonaws.com/donovanbailey/products/api_...   

                                      product_colors  
0  [{'hex_value': '#B28378', 'colour_name': 'BFF ...  
1  [{'hex_value': '#b72227', 'colour_name': 'Bee'...  
2  [{'hex_value': '#F2DEC3', 'colour_name': 'Fair...  
3  [{'hex_value': '#F2DEC3', 'colour_name': 'Fair...

In [6]:
makeup_api.tail() # print tail of dataframe

id               brand  \
926   5            pacifica   
927   4  physicians formula   
928   3  physicians formula   
929   2                None   
930   1          pure anada   

                                                  name  price price_sign  \
926  Pacifica Stellar Gaze Length & Strength Minera...   28.0       None   
927    Physicians Formula Organic Wear FakeOut Mascara  15.79       None   
928  Physicians Formula Organic Wear Lash Boosting ...  15.79       None   
929                  L' Oreal Paris Voluminous Mascara   9.99       None   
930                         Pure Anada Natural Mascara   14.0       None   

    currency                                         image_link  \
926     None  https://d3t32hsnjxo7q6.cloudfront.net/i/b4a5f3...   
927     None  https://d3t32hsnjxo7q6.cloudfront.net/i/2b3d55...   
928     None  https://d3t32hsnjxo7q6.cloudfront.net/i/ca3136...   
929     None  https://d3t32hsnjxo7q6.cloudfront.net/i/4d671a...   
930     None  https://d3t32hsnjxo7q6.cloudfront.net/i/afefed...   

                                          product_link     website_link  \
926  https://well.ca/products/pacifica-stellar-gaze...  https://well.ca   
927  https://well.ca/products/physicians-formula-or...  https://well.ca   
928  https://well.ca/products/physicians-formula-or...  https://well.ca   
929  https://well.ca/products/l-oreal-paris-volumin...  https://well.ca   
930  https://well.ca/products/pure-anada-natural-ma...  https://well.ca   

                                           description  rating category  \
926  You will fall in love with Pacifica's long las...     4.2     None   
927  Physicians Formula Organic Wear FakeOut Mascar...     3.9     None   
928  Physicians Formula Organic Wear Lash Boosting ...     3.7     None   
929  Voluminous - Volume Building Waterproof Mascar...     4.5     None   
930  Finally, a natural mascara you can feel good a...     4.6     None   

    product_type                          tag_list                created_at  \
926      mascara                  [Natural, Vegan]  2016-10-01T18:10:52.756Z   
927      mascara            [Natural, Gluten Free]  2016-10-01T18:10:51.387Z   
928      mascara            [Natural, Gluten Free]  2016-10-01T18:10:50.417Z   
929      mascara                                []  2016-10-01T18:10:49.185Z   
930      mascara  [Natural, Gluten Free, Canadian]  2016-10-01T18:10:46.893Z   

                   updated_at  \
926  2017-12-23T20:31:47.230Z   
927  2017-12-23T20:31:47.075Z   
928  2017-12-23T20:31:46.951Z   
929  2017-12-23T20:31:46.436Z   
930  2017-12-23T20:31:46.013Z   

                                       product_api_url  \
926  http://makeup-api.herokuapp.com/api/v1/product...   
927  http://makeup-api.herokuapp.com/api/v1/product...   
928  http://makeup-api.herokuapp.com/api/v1/product...   
929  http://makeup-api.herokuapp.com/api/v1/product...   
930  http://makeup-api.herokuapp.com/api/v1/product...   

                                    api_featured_image  \
926  //s3.amazonaws.com/donovanbailey/products/api_...   
927  //s3.amazonaws.com/donovanbailey/products/api_...   
928  //s3.amazonaws.com/donovanbailey/products/api_...   
929  //s3.amazonaws.com/donovanbailey/products/api_...   
930  //s3.amazonaws.com/donovanbailey/products/api_...   

                                        product_colors  
926                                                 []  
927  [{'hex_value': '#000000', 'colour_name': 'Ultr...  
928  [{'hex_value': '#000000', 'colour_name': 'Ultr...  
929  [{'hex_value': '#231F20', 'colour_name': 'Blac...  
930  [{'hex_value': '#000000', 'colour_name': 'Blac...

In [7]:
# info about the columns of data 
makeup_api.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 931 entries, 0 to 930
Data columns (total 19 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   id                  931 non-null    int64  
 1   brand               919 non-null    object 
 2   name                931 non-null    object 
 3   price               917 non-null    object 
 4   price_sign          368 non-null    object 
 5   currency            368 non-null    object 
 6   image_link          931 non-null    object 
 7   product_link        931 non-null    object 
 8   website_link        931 non-null    object 
 9   description         930 non-null    object 
 10  rating              340 non-null    float64
 11  category            517 non-null    object 
 12  product_type        931 non-null    object 
 13  tag_list            931 non-null    object 
 14  created_at          931 non-null    object 
 15  updated_at          931 non-null    object 
 16  product_

By looking at the datatype above, datatype for price is 'object' which clearly needs to be float. This shows the shape of the data where we have 931 rows and 19 columns. Also shows that some of the columns have null values.

In [8]:
# changing datatype of price from object to float
makeup_api.price = makeup_api.price.astype(float)
makeup_api.price.dtype

dtype('float64')

In [9]:
# get summary statistics of the data
makeup_api.describe()

id       price      rating
count   931.000000  917.000000  340.000000
mean    531.163265   16.508593    4.319118
std     311.054915   11.028035    0.675849
min       1.000000    0.000000    1.500000
25%     263.000000    8.990000    4.000000
50%     518.000000   13.990000    4.500000
75%     814.500000   22.000000    5.000000
max    1048.000000   77.000000    5.000000

In [10]:
mean = makeup_api['rating'].mean().round(2)
min_ratings = makeup_api['rating'].min()
max_ratings = makeup_api['rating'].max()

print(f'The average ratings for all the makeup brand is {mean}')
print(f'The minimum ratings for all the makeup brand is {min_ratings}')
print(f'The maximum ratings for all the makeup brand is {max_ratings}')


The average ratings for all the makeup brand is 4.32
The minimum ratings for all the makeup brand is 1.5
The maximum ratings for all the makeup brand is 5.0


Get all the rows where the price is greater than and equal to 30 to see the highend makeup products.

In [11]:
highend_makeup = makeup_api[makeup_api['price'] >= 30]
highend_makeup



id            brand                                               name  \
48   1000         glossier                               Perfecting Skin Tint   
201   846         clinique  Even Better&trade; Compact Makeup Broad Spectr...   
204   843         clinique  Repairwear Laser Focus&trade; All-Smooth Makeu...   
219   828         clinique  Clinique BIY&trade; Blend It Yourself Pigment ...   
228   819         clinique  Moisture Surge&trade; CC Cream Hydrating Colou...   
..    ...              ...                                                ...   
734   224            stila                     Stila La Quill Eye Liner Brush   
764   186     dr. hauschka                       Dr. Hauschka Liquid Eyeliner   
840    94  cargo cosmetics                         Cargo Cosmetics ColorStick   
853    80     dr. hauschka                             Dr. Hauschka Lipstick    
904    27     dr. hauschka                               Dr. Hauschka Mascara   

     price price_sign currency  \
48    32.0          $      USD   
201   32.0          $      USD   
204   36.0          $      USD   
219   33.0          $      USD   
228   39.0          $      USD   
..     ...        ...      ...   
734   36.0       None     None   
764   30.0       None     None   
840   32.0       None     None   
853   34.0       None     None   
904   33.0       None     None   

                                            image_link  \
48   https://static-assets.glossier.com/production/...   
201  https://www.clinique.com/media/export/cms/prod...   
204  https://www.clinique.com/media/export/cms/prod...   
219  https://www.clinique.com/media/export/cms/prod...   
228  https://www.clinique.com/media/export/cms/prod...   
..                                                 ...   
734  https://d3t32hsnjxo7q6.cloudfront.net/i/f70d67...   
764  https://d3t32hsnjxo7q6.cloudfront.net/i/a16a04...   
840  https://d3t32hsnjxo7q6.cloudfront.net/i/185545...   
853  https://d3t32hsnjxo7q6.cloudfront.net/i/659877...   
904  https://d3t32hsnjxo7q6.cloudfront.net/i/85c6b5...   

                                          product_link  \
48   https://www.glossier.com/products/perfecting-s...   
201  https://www.clinique.com/product/1599/24867/ma...   
204  https://www.clinique.com/product/1599/19314/ma...   
219  https://www.clinique.com/product/1599/45511/ma...   
228  https://www.clinique.com/product/11480/25255/m...   
..                                                 ...   
734  https://well.ca/products/stila-la-quill-eye-li...   
764  https://well.ca/products/dr-hauschka-liquid-ey...   
840  https://well.ca/products/cargo-cosmetics-color...   
853  https://well.ca/products/dr-hauschka-lipstick_...   
904  https://well.ca/products/dr-hauschka-mascara_3...   

                 website_link  \
48   https://www.glossier.com   
201  https://www.clinique.com   
204  https://www.clinique.com   
219  https://www.clinique.com   
228  https://www.clinique.com   
..                        ...   
734           https://well.ca   
764           https://well.ca   
840           https://well.ca   
853           https://well.ca   
904           https://well.ca   

                                           description  rating  category  \
48   In the land between bare skin and makeup makeu...     NaN      None   
201  A light cream with a powdery finish, our compa...     NaN      None   
204                                                        NaN      None   
219  Transform your favorite moisturizer into a BB,...     NaN      None   
228  One simple step for glowing skin. Lightweight ...     NaN     bb_cc   
..                                                 ...     ...       ...   
734  Stila La Quill Eye Liner Brush is a coveted se...     NaN    liquid   
764  The fine brush on the Liquid Eyeliner, availab...     4.0    liquid   
840  Cargo Cosmetics ColorSticks are an oil-free, c...     4.0  lipstick   
853  Dr. Hauschka Lipstick has all the nurturing qu...     4.9  lipstick  

In [12]:
# info on lowerend makeup 
lowerend_makeup = makeup_api[makeup_api['price'] <= 10]
lowerend_makeup

id       brand                                              name  \
0    1048   colourpop                                     Lippie Pencil   
1    1047   colourpop                                       Blotted Lip   
2    1046   colourpop                                       Lippie Stix   
5    1043      deciem                                  Serum Foundation   
6    1042      deciem                               Coverage Foundation   
..    ...         ...                                               ...   
918    13     l'oreal  L'Oreal Paris Voluminous Volume Building Mascara   
922     9  maybelline      Maybelline Great Lash Lots of Lashes Mascara   
924     7  maybelline      Maybelline Volum'Express the Falsies Mascara   
925     6   covergirl      CoverGirl Lashblast Volume Blasting Mascara    
929     2        None                 L' Oreal Paris Voluminous Mascara   

     price price_sign currency  \
0     5.00          $      CAD   
1     5.50          $      CAD   
2     5.50          $      CAD   
5     6.70          $      CAD   
6     6.90          $      CAD   
..     ...        ...      ...   
918   9.99       None     None   
922   7.79       None     None   
924   9.79       None     None   
925   9.99       None     None   
929   9.99       None     None   

                                            image_link  \
0    https://cdn.shopify.com/s/files/1/1338/0845/co...   
1    https://cdn.shopify.com/s/files/1/1338/0845/pr...   
2    https://cdn.shopify.com/s/files/1/1338/0845/co...   
5    https://3bc01d2807fb1bc0d25c-a86d2521f1af89898...   
6    https://3bc01d2807fb1bc0d25c-a86d2521f1af89898...   
..                                                 ...   
918  https://d3t32hsnjxo7q6.cloudfront.net/i/c9288f...   
922  https://d3t32hsnjxo7q6.cloudfront.net/i/dfaf77...   
924  https://d3t32hsnjxo7q6.cloudfront.net/i/af1f35...   
925  https://d3t32hsnjxo7q6.cloudfront.net/i/18790b...   
929  https://d3t32hsnjxo7q6.cloudfront.net/i/4d671a...   

                                          product_link           website_link  \
0      https://colourpop.com/collections/lippie-pencil  https://colourpop.com   
1    https://colourpop.com/collections/lippie-stix?...  https://colourpop.com   
2        https://colourpop.com/collections/lippie-stix  https://colourpop.com   
5    https://deciem.com/product/rdn-serum-foundatio...    https://deciem.com/   
6    https://deciem.com/product/rdn-coverage-founda...    https://deciem.com/   
..                                                 ...                    ...   
918  https://well.ca/products/loreal-paris-volumino...        https://well.ca   
922  https://well.ca/products/maybelline-great-lash...        https://well.ca   
924  https://well.ca/products/maybelline-volumexpre...        https://well.ca   
925  https://well.ca/products/covergirl-lashblast-v...        https://well.ca   
929  https://well.ca/products/l-oreal-paris-volumin...        https://well.ca   

                                           description  rating  category  \
0    Lippie Pencil A long-wearing and high-intensit...     NaN    pencil   
1    Blotted Lip Sheer matte lipstick that creates ...     NaN  lipstick   
2    Lippie Stix Formula contains Vitamin E, Mango,...     NaN  lipstick   
5    Serum Foundations are lightweight medium-cover...     NaN    liquid   
6    Coverage Foundations are full-coverage formula...     NaN    liquid   
..                                                 ...     ...       ...   
918  4X Fuller Lashes\n\n\nDramatic Volume Effect\n...     2.5      None   
922  Introducing a Great Lash innovation - Great La...     4.2      None   
924  False lash glam in a tube, instantly!The Falsi...     4.0      None   
925  Get a blast of lush, volumized lashes! LashBla...     4.8      None   
929  Voluminous - Volume Building Waterproof Mascar...     4.5      None   

    product_type                                           tag_list  \
0      lip_liner                           

In [13]:
# select and print the second and third columns of the dataframe
makeup_api[['brand', 'name']].head()

brand                  name
0  colourpop         Lippie Pencil
1  colourpop           Blotted Lip
2  colourpop           Lippie Stix
3  colourpop  No Filter Foundation
4      boosh              Lipstick

In [14]:
# Alternative way to get 2nd and 3rd column using iloc method
makeup_api.iloc[:, 1:3]

brand                                               name
0             colourpop                                      Lippie Pencil
1             colourpop                                        Blotted Lip
2             colourpop                                        Lippie Stix
3             colourpop                               No Filter Foundation
4                 boosh                                           Lipstick
..                  ...                                                ...
926            pacifica  Pacifica Stellar Gaze Length & Strength Minera...
927  physicians formula    Physicians Formula Organic Wear FakeOut Mascara
928  physicians formula  Physicians Formula Organic Wear Lash Boosting ...
929                None                  L' Oreal Paris Voluminous Mascara
930          pure anada                         Pure Anada Natural Mascara

[931 rows x 2 columns]

In [17]:
# printing first 4 rows
makeup_api.loc[0:3, :]

id      brand                  name  price price_sign currency  \
0  1048  colourpop         Lippie Pencil    5.0          $      CAD   
1  1047  colourpop           Blotted Lip    5.5          $      CAD   
2  1046  colourpop           Lippie Stix    5.5          $      CAD   
3  1045  colourpop  No Filter Foundation   12.0          $      CAD   

                                          image_link  \
0  https://cdn.shopify.com/s/files/1/1338/0845/co...   
1  https://cdn.shopify.com/s/files/1/1338/0845/pr...   
2  https://cdn.shopify.com/s/files/1/1338/0845/co...   
3  https://cdn.shopify.com/s/files/1/1338/0845/pr...   

                                        product_link  \
0    https://colourpop.com/collections/lippie-pencil   
1  https://colourpop.com/collections/lippie-stix?...   
2      https://colourpop.com/collections/lippie-stix   
3  https://colourpop.com/products/no-filter-matte...   

                                        website_link  \
0                              https://colourpop.com   
1                              https://colourpop.com   
2                              https://colourpop.com   
3  https://colourpop.com/products/no-filter-matte...   

                                         description  rating  category  \
0  Lippie Pencil A long-wearing and high-intensit...     NaN    pencil   
1  Blotted Lip Sheer matte lipstick that creates ...     NaN  lipstick   
2  Lippie Stix Formula contains Vitamin E, Mango,...     NaN  lipstick   
3  Developed for the Selfie Age, our buildable fu...     NaN    liquid   

  product_type               tag_list                created_at  \
0    lip_liner  [cruelty free, Vegan]  2018-07-08T23:45:08.056Z   
1     lipstick  [cruelty free, Vegan]  2018-07-08T22:01:20.178Z   
2     lipstick  [cruelty free, Vegan]  2018-07-08T21:47:49.858Z   
3   foundation  [cruelty free, Vegan]  2018-07-08T18:22:25.273Z   

                 updated_at  \
0  2018-07-09T00:53:23.301Z   
1  2018-07-09T00:53:23.287Z   
2  2018-07-09T00:53:23.274Z   
3  2018-07-09T00:53:23.313Z   

                                     product_api_url  \
0  http://makeup-api.herokuapp.com/api/v1/product...   
1  http://makeup-api.herokuapp.com/api/v1/product...   
2  http://makeup-api.herokuapp.com/api/v1/product...   
3  http://makeup-api.herokuapp.com/api/v1/product...   

                                  api_featured_image  \
0  //s3.amazonaws.com/donovanbailey/products/api_...   
1  //s3.amazonaws.com/donovanbailey/products/api_...   
2  //s3.amazonaws.com/donovanbailey/products/api_...   
3  //s3.amazonaws.com/donovanbailey/products/api_...   

                                      product_colors  
0  [{'hex_value': '#B28378', 'colour_name': 'BFF ...  
1  [{'hex_value': '#b72227', 'colour_name': 'Bee'...  
2  [{'hex_value': '#F2DEC3', 'colour_name': 'Fair...  
3  [{'hex_value': '#F2DEC3', 'colour_name': 'Fair...

In [18]:
makeup_api.head(4)

id      brand                  name  price price_sign currency  \
0  1048  colourpop         Lippie Pencil    5.0          $      CAD   
1  1047  colourpop           Blotted Lip    5.5          $      CAD   
2  1046  colourpop           Lippie Stix    5.5          $      CAD   
3  1045  colourpop  No Filter Foundation   12.0          $      CAD   

                                          image_link  \
0  https://cdn.shopify.com/s/files/1/1338/0845/co...   
1  https://cdn.shopify.com/s/files/1/1338/0845/pr...   
2  https://cdn.shopify.com/s/files/1/1338/0845/co...   
3  https://cdn.shopify.com/s/files/1/1338/0845/pr...   

                                        product_link  \
0    https://colourpop.com/collections/lippie-pencil   
1  https://colourpop.com/collections/lippie-stix?...   
2      https://colourpop.com/collections/lippie-stix   
3  https://colourpop.com/products/no-filter-matte...   

                                        website_link  \
0                              https://colourpop.com   
1                              https://colourpop.com   
2                              https://colourpop.com   
3  https://colourpop.com/products/no-filter-matte...   

                                         description  rating  category  \
0  Lippie Pencil A long-wearing and high-intensit...     NaN    pencil   
1  Blotted Lip Sheer matte lipstick that creates ...     NaN  lipstick   
2  Lippie Stix Formula contains Vitamin E, Mango,...     NaN  lipstick   
3  Developed for the Selfie Age, our buildable fu...     NaN    liquid   

  product_type               tag_list                created_at  \
0    lip_liner  [cruelty free, Vegan]  2018-07-08T23:45:08.056Z   
1     lipstick  [cruelty free, Vegan]  2018-07-08T22:01:20.178Z   
2     lipstick  [cruelty free, Vegan]  2018-07-08T21:47:49.858Z   
3   foundation  [cruelty free, Vegan]  2018-07-08T18:22:25.273Z   

                 updated_at  \
0  2018-07-09T00:53:23.301Z   
1  2018-07-09T00:53:23.287Z   
2  2018-07-09T00:53:23.274Z   
3  2018-07-09T00:53:23.313Z   

                                     product_api_url  \
0  http://makeup-api.herokuapp.com/api/v1/product...   
1  http://makeup-api.herokuapp.com/api/v1/product...   
2  http://makeup-api.herokuapp.com/api/v1/product...   
3  http://makeup-api.herokuapp.com/api/v1/product...   

                                  api_featured_image  \
0  //s3.amazonaws.com/donovanbailey/products/api_...   
1  //s3.amazonaws.com/donovanbailey/products/api_...   
2  //s3.amazonaws.com/donovanbailey/products/api_...   
3  //s3.amazonaws.com/donovanbailey/products/api_...   

                                      product_colors  
0  [{'hex_value': '#B28378', 'colour_name': 'BFF ...  
1  [{'hex_value': '#b72227', 'colour_name': 'Bee'...  
2  [{'hex_value': '#F2DEC3', 'colour_name': 'Fair...  
3  [{'hex_value': '#F2DEC3', 'colour_name': 'Fair...